### PR TITLE
core: do not subscribe to new connections from iOS NSE (subscribe=off flag), subscribe in app when it activates

### PR DESCRIPTION
**Pull request by dpwiz**
Client counterpart of https://github.com/simplex-chat/simplexmq/pull/839

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -565,10 +572,11 @@ processChatCommand = \case
             smpSndFileTransfer :: CryptoFile -> Integer -> Maybe InlineFileMode -> m (FileInvitation, CIFile 'MDSnd, FileTransferMeta)
             smpSndFileTransfer (CryptoFile _ (Just _)) _ _ = throwChatError $ CEFileInternal "locally encrypted files can't be sent via SMP" -- can only happen if XFTP is disabled
             smpSndFileTransfer (CryptoFile file Nothing) fileSize fileInline = do
+              subMode <- chatReadVar subscriptionMode
               (agentConnId_, fileConnReq) <-
                 if isJust fileInline
                   then pure (Nothing, Nothing)
-                  else bimap Just Just <$> withAgent (\a -> createConnection a (aUserId user) True SCMInvitation Nothing)
+                  else bimap Just Just <$> withAgent (\a -> createConnection a (aUserId user) True subMode SCMInvitation Nothing)
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:579]: needs to move to the chat

File: src/Simplex/Chat.hs [hunk 2]
```diff
@@ -462,14 +465,18 @@ processChatCommand = \case
   APIActivateChat -> withUser $ \_ -> do
     restoreCalls
     withAgent foregroundAgent
-    withStoreCtx' (Just "APIActivateChat, getUsers") getUsers >>= void . forkIO . startFilesToReceive
+    users <- withStoreCtx' (Just "APIActivateChat, getUsers") getUsers
+    chatReadVar subscriptionMode >>= \case
+      SMOnlyCreate -> void . forkIO $ subscribeUsers True users
+      SMSubscribe -> pure () -- Subscriptions are automatic, nothing to do
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:471]:
```suggestion
    chatReadVar subscriptionMode >>= \case
      SMOnlyCreate -> pure ()
      SMSubscribe -> void . forkIO $ subscribeUsers True users
```

It's actually other way around, you want to subscribe new subscriptions only in SMSubscribe mode.

File: src/Simplex/Chat.hs [hunk 3]
```diff
@@ -244,6 +245,8 @@ cfgServers = \case
 startChatController :: forall m. ChatMonad' m => Bool -> Bool -> Bool -> m (Async ())
 startChatController subConns enableExpireCIs startXFTPWorkers = do
   asks smpAgent >>= resumeAgentClient
+  unless subConns $
+    chatWriteVar subscriptionMode SMOnlyCreate
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:249]:
```suggestion
  unless subConns $ chatWriteVar subscriptionMode SMOnlyCreate
```

File: src/Simplex/Chat.hs [hunk 4]
```diff
@@ -2413,18 +2431,28 @@ agentSubscriber = do
...
+        (gs, mConns, ms) <- getGroupMemberConns
+        (sftConns, sfts) <- getSndFileTransferConns
+        (rftConns, rfts) <- getRcvFileTransferConns
+        (pcConns, pcs) <- getPendingContactConns
+        let conns = concat [ctConns, ucConns, mConns, sftConns, rftConns, pcConns]
+        pure (conns, cts, ucs, gs, ms, sfts, rfts, pcs)
   -- subscribe using batched commands
-  rs <- withAgent (`agentBatchSubscribe` concat [ctConns, ucConns, mConns, sftConns, rftConns, pcConns])
+  rs <- withAgent $ \a -> agentBatchSubscribe a conns
+  -- when onlyNeeded $ withStore' (`unsetConnectionNeedsSub` conns)
```
Comment (spaced4ndy) [inline src/Simplex/Chat.hs:2455]:
```suggestion
```

**Pull request merged**